src/queue.rs

```diff
@@ -268,6 +205,31 @@ impl<M: GuestAddressSpace> DescriptorChain<M> {
             writable: true,
         }
     }
+
+    // Alters the internal state of the `DescriptorChain` to switch iterating over an
+    // indirect descriptor table defined by `desc`.
+    fn process_indirect_descriptor(&mut self, desc: Descriptor) -> Result<(), Error> {
+        if self.is_indirect {
+            return Err(Error::InvalidIndirectDescriptor);
+        }
+
+        let table_len = (desc.len as usize) / VIRTQ_DESCRIPTOR_SIZE;
+        // Check the target indirect descriptor table is correctly aligned.
+        if desc.addr().raw_value() & (VIRTQ_DESCRIPTOR_SIZE as u64 - 1) != 0
+            || (desc.len as usize) & (VIRTQ_DESCRIPTOR_SIZE - 1) != 0
+            || table_len > usize::from(std::u16::MAX)
+        {
+            return Err(Error::InvalidIndirectDescriptor);
```

Comment:
I would differentiate between this case and the one a few lines higher with a separate `Error`.